### ISSUE TITLE
Implementing Index on serde_json::Map

### ISSUE THREAD

**Issue by roblabla** [serde-rs/json#217]
Indexing on Value is very nice ! However, it'd be even nicer if serde_json::Map could be indexed in a similar way. I have some code where I match over a Value that looks like this :

```rust
match *val {
    Value::String(ref s) => s,
    Value::Array(ref arr) => arr[0].as_str().unwrap(),
    Value::Object(_) => val["type"].as_str().unwrap(),
    _ => panic!()
}
```

It'd be a bit nicer if I could instead write

```rust
match *val {
    Value::String(ref s) => s,
    Value::Array(ref arr) => arr[0].as_str().unwrap(),
    Value::Object(ref map) => map["type"].as_str().unwrap(),
    _ => panic!()
}
```

**Comment by dtolnay** [comment]
Makes sense. I opened #218 to add this.

**Comment by dtolnay** [comment]
@roblabla I released this in serde_json 0.9.2.
